### PR TITLE
Casmuser 2720 Add Broker Shared Secret Managment to UAS

### PR DESCRIPTION
**Pull request by erl-hpe**
## Summary and Scope

Add support for shared SSH credentials for Broker UAI replicas to UAS:

- Code to generate a new environment variable UAI_SHARED_SECRET_PATH for any UAI that has a configured UAI Creation Class when it is used to make a UAI (i.e. Broker UAIS)
- Code to generate a new environment variable UAI_REPLICAS for all UAIs telling them how many replicas their class is configured for (this is mostly a convenience for some of the backward compatibility stuff in switchboard, but it may be handy in the future for other things)
- Code to clean up secrets when a UAI class is deleted (this is implemented as best-effort, don't worry if it doesn't work, code since that is really all we can do here)
- New unit tests for the new vault code above

This is a backward compatible change.

## Issues and Related PRs

* Future work required by [CASMUSER-2720](https://jira-pro.its.hpecorp.net:8443/browse/CASMUSER-2720)

### Tested on:

  * Virtual Shasta

### Test description:

Deployed on virtual shasta and created 2 different Broker UAI Classes configured to create End-User UAIs using 2 different End-User UAI classes.  Deployed both classes of Broker UAI and logged into both to verify that End-User UAIs were created for both and that I was permitted to log into both UAIs seamlessly.  Verified that this worked with both single replica and multi-replica Broker UAIs.  Used `kubectl exec` to examine the environment variables provided by the UAS when the Broker UAIs were created and verified that the correct information was provided.  Used `curl` to log into and query vault for the shared secrets to ensure that the shared secrets were stored in the correct place in vault.  Verified that the Broker UAIs used the vault shared secrets (as opposed by locally stored secrets).  Deleted the creation class for one of the Broker UAIs and used `curl` to examine vault and verify that the secrets for the deleted class were removed but the secrets for the still existing class were left untouched, as were other unrelated secrets stored in vault.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? No
- Were continuous integration tests run? If not, why? N/A
- Was upgrade tested? If not, why? Yes
- Was downgrade tested? If not, why? Yes
- Were new tests (or test issues/Jiras) created for this change? Unit tests only

## Risks and Mitigations

There are no known risks associated with this change.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [ N/A ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -5,7 +5,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-## [1.18.0] - 2021-12-15
+## [1.19.0] - 2022-02-24
+- Add logic for managing Broker UAI shared secrets
+
+## [1.18.0] - 2022-02-01
```
Comment (erl-hpe) [inline ChangeLog.md:11]: REVIEWERS: fixed the date on this, it was incorrect.

File: requirements.txt [hunk 2]
```diff
@@ -6,4 +6,4 @@ requests == 2.25.1
 sshpubkeys >= 3.1.0
 etcd3_model >= 1.0.0
 flask < 2.0.0
-
+markupsafe==2.0.1
```
Comment (erl-hpe) [inline requirements.txt:9]: REVIEWERS:  In going from 2.0.1 to 2.1.0 markupsafe removed a function from the library that flask depends on.  This addresses that until we can work on updating to handle new dependencies.

File: swagger_server/uas_lib/uai_mgr.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # MIT License
 #
-# (C) Copyright [2020] Hewlett Packard Enterprise Development LP
+# (C) Copyright [2022] Hewlett Packard Enterprise Development LP
```
Comment (alanm-hpe) [inline swagger_server/uas_lib/uai_mgr.py:3]: Shouldn't we want 2020-2022 here?

**Review by alanm-hpe — DISMISSED** [review]

File: swagger_server/uas_lib/uai_mgr.py (hunk 3 above)
Comment (erl-hpe) [reply swagger_server/uas_lib/uai_mgr.py:3]: Yeah, we should.  I will fix that.  Thanks for catching it.

**Review by alanm-hpe — APPROVED** [review]

**Pull request merged**